### PR TITLE
refactor: rename Argo CD headless feature to Core

### PR DESCRIPTION
**Pull request by alexmt**
Signed-off-by: Alexander Matyushentsev <AMatyushentsev@gmail.com>

As discussed PR renames Argo CD headless feature to Core.

Instead of using `argocd login --headless` user should use `argocd login --k8s-api`

### CODE REVIEW

File: docs/operator-manual/installation.md [hunk 1]
```diff
@@ -46,15 +46,15 @@ High Availability installation is recommended for production use. Bundle include
 * [ha/namespace-install.yaml](https://github.com/argoproj/argo-cd/blob/master/manifests/ha/namespace-install.yaml) - the same as namespace-install.yaml but
   with multiple replicas for supported components.
 
-## Single-Tenant (a.k.a. "Headless")
+## Single-Tenant (a.k.a. "Core")
 
-The single-tenant installation is most suitable for cluster administrators who indepently use Argo CD and don't need multi-tennancy features. This installation
+The single-tenant/core installation is most suitable for cluster administrators who indepently use Argo CD and don't need multi-tennancy features. This installation
```
Comment (jessesuen) [inline docs/operator-manual/installation.md:51]: I think I would rename the title of the section from "single-tenant (AKA core)" to just "Core" installation. The fact that this is useful for single-tenancy can be mentioned in the description of when you would want to use this mode of installation.

typo: tenancy

File: docs/roadmap.md [hunk 2]
```diff
@@ -42,10 +42,10 @@ via Git repository.
 The feature streamlines the process of adding repositories and clusters to the project and makes it self-service.
 Instead of asking an administrator to change Argo CD settings end users can perform the change independently.
 
-### Headless Argo CD (aka GitOps Agent) ([proposal](https://github.com/argoproj/argo-cd/pull/6385))
+### Core Argo CD (aka GitOps Agent) ([proposal](https://github.com/argoproj/argo-cd/pull/6385))
```
Comment (jessesuen) [inline docs/roadmap.md:45]: Let's remove "GitOps agent". I dont think we want to use that terminology yet.

File: docs/operator-manual/installation.md (hunk 1 above)
Comment (alexmt) [reply docs/operator-manual/installation.md]: agree - fixed.

File: docs/roadmap.md (hunk 2 above)
Comment (alexmt) [reply docs/roadmap.md]: done

**Review by jessesuen — APPROVED** [review]

**Pull request merged**